glass/lensing.py

```diff
@@ -442,7 +442,7 @@ def add_window(self, delta: npt.NDArray[np.float64], w: RadialWindow) -> None:
     def add_plane(
         self,
         delta: npt.NDArray[np.float64],
-        zsrc: float,
+        zsrc: float | npt.NDArray[np.float64],
```

Comment:
If `zeff` can be an array, `zsrc` in `add_window` can be one too, and `add_window` passes `zsrc` to `add_plane`